Add SingleProductBackupBody Component

robertf4 opened the pull request:
#### Changes proposed in this Pull Request:
* This adds a SingleProductBackupBody component that will be used in #14018.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
See the P2 here for the design of this PR: p1HpG7-7MK-p2 (specifically the part titled "WP Admin Desktop")
See the P2 here for the overall MT: p1HpG7-7ET-p2

#### Testing instructions:
* This cannot be tested on its own here. Use #14018 to test.

#### Proposed changelog entry for your changes:
* No changelog entry needed.

robertf4 committed:
Add products endpoint

robertf4 committed:
Add QueryProducts data component and corresponding Redux and rest API calls

robertf4 committed:
[not verified] Add error code

robertf4 committed:
Update _inc/lib/class.core-rest-api-endpoints.php

Co-Authored-By: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

robertf4 committed:
Update _inc/lib/class.core-rest-api-endpoints.php

Co-Authored-By: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

robertf4 committed:
Update _inc/client/state/site/reducer.js

Co-Authored-By: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

robertf4 committed:
[not verified] Add SingleProductBackupBody component

delawski commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -35,6 +42,83 @@ export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
+				</div>
+
+				<div className="single-product-backup__upgrade-button-container">
+					<Button href={ upgradeLinks[ selectedBackupType ] } primary>
```
We're not checking if the `upgradeLinks` exists and if it contains a member we're trying to access. In our `propTypes` definition we're also not making it a required prop, so I guess we may run into an issue here.

The same apply to the next line where we're accessing the `upgradeTitles` object.

jeherve replied:
Done in 9cdc47884a5871247d380f18bcf938800d1d276a

delawski commented on _inc/client/plans/single-product-backup.scss:
```diff
@@ -1,3 +1,15 @@
+	h4 {
+		font-size: 14px;
+		color: #747474;
+		border-bottom: solid 2px #e6e6e6;
```
I think that the figma design says the border-width should be 1px.

robertf4 replied:
I'm pretty sure it's 2px, I checked in the Figma both by zooming in to count pixels and looking at the property value.

delawski replied:
Maybe we're using different Figma files for reference? Here's a screenshot of what I can see in the "Backups Single Product Only" project:

![Screenshot 2019-11-15 at 10 54 00](https://user-images.githubusercontent.com/478735/68934088-5160ab00-0796-11ea-9694-4d5581fdae4f.png)

In any case, we can leave it as is. In case the designers want to change it to 1px, we can do it easily later on.

delawski reviewed (changes requested):
@robertf4 This looks good in general. I left some minor comments, please have a look.

jeherve commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -35,6 +42,83 @@ export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
+	constructor() {
+		super();
+
+		this.handleSelectedBackupTypeChange = this.handleSelectedBackupTypeChange.bind( this );
```
Do you think it'd be possible to drop binding and switch to using class properties / arrow functions instead?

jeherve replied:
Done in 4ca5cfbb24eb5d39b03847145d9bb532797c3f55

jeherve commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -35,6 +42,83 @@ export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
+						'Always-on backups ensure you never lose your site. Choose from real-time or daily backups. {{a}}Which one do I need?{{/a}}',
+						{
+							components: {
+								a: <ExternalLink href="https://jetpack.com/upgrade/backup/" icon iconSize={ 12 } />,
```
Do we need to add tracking to that link, maybe via `getUpgradeUrl`?

robertf4 replied:
Good call, but let's do it in another PR after I get the redirects into wpcom so that we can still E2E test this.

jeherve replied:
Logged in #14049

jeherve commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -35,6 +42,83 @@ export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
+				</div>
+
+				<div className="single-product-backup__upgrade-button-container">
+					<Button href={ upgradeLinks[ selectedBackupType ] } primary>
```
Should we add tracking to all those buttons? We have `analytics.tracks.recordJetpackClick` that can help you there, as I think ideally we'd want to be able to follow where people are going form there.

robertf4 replied:
Also good to do in another PR.

jeherve replied:
Logged in #14049

jeherve commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -35,6 +42,83 @@ export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
+				<h4>{ __( 'Backup options:' ) }</h4>
+				<div className="single-product-backup__radio-buttons-container">
+					{ backupOptions.map( option => (
+						<PlanRadioButton
```
Do we need a `key` attribute here to avoid warnings?

robertf4 replied:
Yes, fixed.

jeherve replied:
Done in 237176521064985cb78c28a85be2f7e1ba6f7559. Marking this as resolved.

robertf4 committed:
[not verified] Rename plans-section__body to single_product_backup__body

robertf4 committed:
[not verified] Give class name to h4 tag

robertf4 committed:
[not verified] Remove single_product_backup__body

robertf4 commented on _inc/client/plans/single-product-backup.scss:
```diff
@@ -1,3 +1,15 @@
+		border-bottom: solid 2px #e6e6e6;
+	}
+
+	.gridicons-external {
```
In the end this class wasn't needed because the gridicon color is handled by the `ExternalLink` component.

robertf4 committed:
[not verified] Check for upgradeLinks and upgradeTitles before using them

robertf4 committed:
[not verified] Switch function style

robertf4 committed:
[not verified] Add key to PlanRadioButton

robertf4 committed:
Remove duplicate definition of getProducts

delawski commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -51,6 +58,80 @@ export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
+						/>
+					) ) }
+				</div>
+				{ upgradeLinks &&
```
Thanks for adding those checks.

I think we still need to check if the `upgradeTitles` object exists first, just like we check for `upgradeLinks`.

What do you think about extracting this whole piece to a separate method like `renderUpgradeLink`. Inside of it, we wouldn't have to build such a long inline condition like we have here. Instead we could just have regular `if` statements that would `return null` if we have no links or titles. Feel free to refactor if you find it useful, it's not a blocker though.

gravityrail replied:
upgradeTitles is always going to be defined because it's set to the following value at the top of the function:

```javascript
const upgradeTitles = {
	'real-time': __( 'Upgrade to Real-Time Backups' ),
	daily: __( 'Upgrade to Daily Backups' ),
};
```

delawski reviewed (approved):
I left one minor comment in your code. Other than that the PR seems good to me so I'm approving.

Feel free to merge after addressing this one issue I commented on and after getting a thumbs-up from @jeherve.

gravityrail committed:
Allow TOS agreement before Jetpack is fully active so we track… (#14041)

* Allow TOS agreement before Jetpack is fully active so we track the connection flow

* Update packages/terms-of-service/src/class-terms-of-service.php

Co-Authored-By: Derek Smart <smart@automattic.com>

* Also update the action that is being called in jetpack

We renamed the action lets also rename it in Jetpack.

* [not verified] Revert "Also update the action that is being called in jetpack"

This reverts commit e9c0f21d51151900e218e8253832ab1a5245575e.


Co-authored-by: Brandon Kraft <public@brandonkraft.com>
Co-authored-by: Enej Bajgoric <enej.bajgoric@gmail.com>

zinigor committed:
Removed Jetpack references in the IXR client. (#14046)

This change allows the package to be used outside of Jetpack, relying on the methods provided by the Manager, plus using the WP_Error class instead of the Jetpack_Error wrapper.

jeherve commented on _inc/client/plans/single-product-backup.scss:
```diff
@@ -1,3 +1,15 @@
+.plans-section__body {
+	h4 {
```
Done in a2cfae22a063b3ccf9c0ee9e8c521273fc914a3d

jeherve commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -51,6 +58,80 @@ export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
+			<React.Fragment>
+				<p>
+					{ __(
+						'Always-on backups ensure you never lose your site. Choose from real-time or daily backups. {{a}}Which one do I need?{{/a}}',
```
Wording is a bit odd here. We go from *you* to *I*. Not a blocker though, leaving that one for Editorial, maybe we can do a final editorial review before launch.

enejb replied:
yes lets do that. It should really be the same betwee calypso and WPadmin

jeherve reviewed (approved):
This looks good to me, barring @delawski's comment and my question about wording. Both can be addressed in a follow-up PR though. It should be good to merge.

jeherve commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -51,6 +58,80 @@ export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
+						'Always-on backups ensure you never lose your site. Choose from real-time or daily backups. {{a}}Which one do I need?{{/a}}',
+						{
+							components: {
+								a: <ExternalLink href="https://jetpack.com/upgrade/backup/" icon iconSize={ 12 } />,
```
It seems that just using `ExternalLink` is not enough. You'll need to add `target="_blank"` for it to really open in a new window.

lezama committed:
Full Sync: Don't allow more than one request to enqueue (#14039)

ChaosExAnima committed:
Spell checking CI integration (#13992)

* Adds spell checking and fixes files.

This implements a package to spellcheck files, and
does a pass to ensure everything is green.

* [not verified] Implements Travis check for spelling

Adds Yarn script and Travis CI for spell checking.

* [not verified] Fixes feedback.

Updates docs to prevent spelling exceptions for certain files.

gravityrail committed:
Merge branch 'master' into add/single-product-backup-body-component

gravityrail committed:
Export SingleProductBackupBody class

gravityrail committed:
[not verified] No need to export single product backup body

tyxla commented:
I'm kind of concerned that we're merging these in branches different from `master`. What's the plan for moving them to `master`? I thought the plan was to have them based on `master` before merging.

jeherve commented:
It will be merged to `master` in #14056.

tyxla commented:
Awesome, thanks!
